core/shared/src/main/scala/fs2/concurrent/Queue.scala

```diff
@@ -21,7 +21,7 @@ trait Enqueue[F[_], A] {
     * Enqueues each element of the input stream to this queue by
     * calling `enqueue1` on each element.
     */
-  def enqueue: Pipe[F, A, Unit] = _.evalMap(enqueue1)
+  def enqueue: Pipe[F, A, INothing] = _.foreach(enqueue1)
```

Comment:
There is a fairly unambiguous "number of expected units here", so I don't see why we need to remove useful combinators by draining